clustermesh-apiserver/etcdinit/root.go

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package etcdinit
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"syscall"
+	"time"
+
+	einit "github.com/cilium/cilium/pkg/kvstore/etcdinit"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/spf13/cobra"
+	client "go.etcd.io/etcd/client/v3"
+)
+
+var (
+	log                     = logging.DefaultLogger.WithField(logfields.LogSubsys, "etcdinit")
+	etcdBinaryLocation      = "/usr/bin/etcd"
+	etcdDataDir             = ""
+	etcdInitialClusterToken = ""
+	etcdClusterName         = ""
+	ciliumClusterName       = ""
+	debug                   = false
+	timeout                 = time.Second
+)
+
+func NewCmd() *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:   "etcdinit",
+		Short: "Initialise an etcd data directory for use by the etcd sidecar of clustermesh-apiserver",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := InitEtcdLocal()
+			// the error has already been handled and logged by InitEtcdLocal. We just use it to determine the exit code
+			if err != nil {
+				os.Exit(-1)
+			}
+		},
+	}
+	rootCmd.Flags().StringVar(&etcdDataDir, "etcd-data-dir", "/var/run/etcd", "Etcd data directory. Should have read/write permissions here.")
+	rootCmd.Flags().StringVar(&etcdInitialClusterToken, "ectd-initial-cluster-token", "clustermesh-apiserver", "Etcd initial cluster token. Used to prevent accidentally joining other etcd clusters that are reachable on the same L2 network domain")
+	rootCmd.Flags().StringVar(&etcdClusterName, "etcd-cluster-name", "clustermesh-apiserver", "Name of the etcd cluster. Must match what etcd is later started with.")
+	rootCmd.Flags().StringVar(&ciliumClusterName, "cluster-name", "clustermesh-apiserver", "Name of the Cilium cluster. This is as distinct from the etcd cluster's name.")
+	rootCmd.Flags().DurationVar(&timeout, "timeout", time.Minute*2, "How long to wait for operations before exiting")
+	rootCmd.Flags().BoolVar(&debug, "debug", false, "Debug log output")
```

Comment:
I forgot to mention earlier. Almost all other commands also supporting setting the flags through environment variables in addition to command line flags. It seems fine to keep this as of now though, to avoid unnecessarily increasing the complexity. We don't expect them to be modified by users in any case.

Reply:
I've implemented this via Viper.